framework/src/st_things/things_stack/framework/things_data_manager.c

```diff
@@ -1154,13 +1153,14 @@ static int parse_resource_type_json_with_internal(cJSON *resource_types_user)
 	struct st_resource_type_s *restype = NULL;
 	int internal_resource_cnt = 0;
 	int internal_resource_type_cnt = 0;
+	cJSON *json_internal_root = NULL;
```

Comment:
Any specific reason for separation?

Reply:
It was given warning that 'json_internal_root' may be used uninitialized in this function at line 1259

Reply:
If we add return statement for NULL "resource_types_user", then there is no need for this separation, right?

Reply:
checking return statement for NULL "resource_types_user". I s already there.